libraries/metrics/include/metrics/metrics_group.hpp

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <prometheus/gauge.h>
+#include <prometheus/registry.h>
+
+#include <iostream>
+
+namespace metrics {
+
+using MetricGetter = std::function<double()>;
+using MetricUpdater = std::function<void()>;
+
+/**
+ * @brief add method that is setting specific gauge metric.
+ */
+#define ADD_GAUGE_METRIC(method, name, description)                                                  \
```

Comment:
any other way how to solve this ? Using defines in modern C++ is pretty much avoided if possible

Reply:
Yes it isn't perfect. But it seems to me that this is better than copy-pasting a lot of code for every metric. Another option is to store some string->label mapping, so we can call some set method with the string. But metrics won't be strictly defined in this case. Small example. This code 
```cpp
  ADD_GAUGE_METRIC_WITH_UPDATER(setPeersCount, "peers_count", "Count of peers that node is connected to")
```
will become this for almost every metric
```cpp
void setPeersCount(double v) { 
  static auto& label = addMetric<prometheus::Gauge>(group_name + "_" +"peers_count",  "Count of peers that node is connected to").Add({});
  label.Set(v);
}
void setPeersCountUpdater(MetricGetter getter) {
  updaters_.push_back([this, getter]() { setPeersCount(getter()); });
}
```